### ISSUE TITLE
`lmplot` fails if any group has only one record

### ISSUE THREAD

**Issue by deeenes** [mwaskom/seaborn#1949]
For example:

```
import numpy as np
import pandas as pd
import seaborn as sns

df = pd.DataFrame({
    'col': np.append(np.random.choice(np.array(['a', 'b', 'c']), 10), ['d']),
    'x': np.random.normal(size = 11),
    'y': np.random.normal(size = 11),
})

sns.lmplot(x = 'x', y = 'y', col = 'col', data = df)
```

It raises `IndexError: invalid index to scalar variable` which is not very intuitive. I think this case can occur with any data coming from the real world, and would be much nicer to exclude this group automatically or simply plot the single dot without regression line.

**Comment by mwaskom** [comment]
Thanks for the reproducible report.

You would think it's a problem fitting a regression with 1 datapoint, but actually if you read the full stack trace (which would be useful to add to the issue), it's occuring in the na-handling code.

**Comment by NamanJain2050** [comment]
Yes @mwaskom you are correct. It's occurring in na-handling code. Can you elaborate a little more why it is stuck there?

**Comment by mwaskom** [comment]
Fixed by #1969
